wallet/src/controller.rs

```diff
@@ -677,20 +677,27 @@ fn create_ok_response(json: &str) -> Response<Body> {
 			"access-control-allow-headers",
 			"Content-Type, Authorization",
 		)
+		.header(hyper::header::CONTENT_TYPE, "application/json")
 		.body(json.to_string().into())
 		.unwrap()
 }
 
 fn response<T: Into<Body>>(status: StatusCode, text: T) -> Response<Body> {
-	Response::builder()
+	let mut builder = &mut Response::builder();
+
+	builder = builder
 		.status(status)
 		.header("access-control-allow-origin", "*")
 		.header(
 			"access-control-allow-headers",
 			"Content-Type, Authorization",
-		)
-		.body(text.into())
-		.unwrap()
+		);
+
+	if status == StatusCode::OK {
+		builder = builder.header(hyper::header::CONTENT_TYPE, "application/json");
+	}
+
+	builder.body(text.into()).unwrap()
```

Comment:
Perhaps add one line of comment here to remind the caller of this `response` function, the `text` here is required to be a JSON format.

Or even we can change the variable name `text` here as for example `json_formated_text`?